### ISSUE TITLE
[pyyaml 3.13] TypeError while dumping lambda in Python 3

### ISSUE THREAD

**Issue by NiklasRosenstein** [yaml/pyyaml#222]
```python
import yaml
print(yaml.dump(lambda: None))
```

This fails with a TypeError in Python 3 (not in Python 2).

```
[...]
  File "/Users/nrosenstein/Projects/.venv/lib/python3.7/site-packages/yaml/emitter.py", line 228, in expect_document_root
    self.expect_node(root=True)
  File "/Users/nrosenstein/Projects/.venv/lib/python3.7/site-packages/yaml/emitter.py", line 242, in expect_node
    self.process_tag()
  File "/Users/nrosenstein/Projects/.venv/lib/python3.7/site-packages/yaml/emitter.py", line 489, in process_tag
    self.prepared_tag = self.prepare_tag(tag)
  File "/Users/nrosenstein/Projects/.venv/lib/python3.7/site-packages/yaml/emitter.py", line 607, in prepare_tag
    for ch in data:
TypeError: ord() expected string of length 1, but int found
```

For reference:

https://github.com/yaml/pyyaml/blob/a2d481b8dbd2b352cb001f07091ccf669227290f/lib3/yaml/emitter.py#L569-L575

*On a side note, I don't know how the line numbers don't match. I have PyYAML 3.13 installed and this references the commit on the 3.13 tag*.

`str.encode()` returns a `bytes` object in Python 3, which in turn yields integers and no substrings when iterated over, leading `ord()` to fail.

**Comment by perlpunk** [comment]
Thanks, this should actually be fixed by #129

**Comment by fersarr** [comment]
We are being hit by this too. I guess this explains the issue https://stackoverflow.com/questions/19897209/troubleshooting-typeerror-ord-expected-string-of-length-1-but-int-found

Any ETA on when #129 will be merged?

**Comment by perlpunk** [comment]
This was fixed in PyYAML 5.1 with #129
